framework/encode/vulkan_state_writer.cpp

```diff
@@ -914,6 +951,168 @@ void VulkanStateWriter::WriteDescriptorSetState(const VulkanStateTable& state_ta
+{
+    assert(asset_file_stream_ != nullptr);
+
+    std::set<util::MemoryOutputStream*> processed;
+
+    std::unordered_map<format::HandleId, const util::MemoryOutputStream*> temp_ds_layouts;
+
+    // First pass over descriptor set table to determine which dependencies need to be created temporarily.
+    state_table.VisitWrappers([&](const vulkan_wrappers::DescriptorSetWrapper* wrapper) {
+        assert(wrapper != nullptr);
+
+        auto ds_layout_wrapper = state_table.GetDescriptorSetLayoutWrapper(wrapper->set_layout_dependency.handle_id);
+        if (ds_layout_wrapper == nullptr)
+        {
+            // The object no longer exists, so a temporary object must be created.
+            auto        dep_create_parameters = wrapper->set_layout_dependency.create_parameters.get();
+            const auto& dep_inserted =
+                temp_ds_layouts.insert(std::make_pair(wrapper->set_layout_dependency.handle_id, dep_create_parameters));
+
+            // Create a temporary object on first encounter.
+            if (dep_inserted.second)
+            {
+                if (wrapper->dirty)
+                {
+                    const int64_t offset                       = asset_file_stream_->GetOffset();
+                    (*asset_file_offsets_)[wrapper->handle_id] = offset;
+                    WriteFunctionCall(
+                        wrapper->set_layout_dependency.create_call_id, dep_create_parameters, asset_file_stream_);
+                    if (output_stream_ != nullptr)
+                    {
+                        WriteExecuteFromFile(1, offset);
+                    }
+                }
+                else
+                {
+                    if (output_stream_ != nullptr)
+                    {
+                        assert((*asset_file_offsets_).find(wrapper->handle_id) != (*asset_file_offsets_).end());
+                        const int64_t offset = (*asset_file_offsets_)[wrapper->handle_id];
+                        WriteExecuteFromFile(1, offset);
+                    }
+                }
+            }
+        }
+    });
+
+    state_table.VisitWrappers([&](vulkan_wrappers::DescriptorSetWrapper* wrapper) {
+        assert(wrapper != nullptr);
+
+        uint32_t n_blocks = 0;
+        int64_t  offset;
+        if (wrapper->dirty)
+        {
+            offset = asset_file_stream_->GetOffset();
+        }
+        else
+        {
+            assert((*asset_file_offsets_).find(wrapper->handle_id) != (*asset_file_offsets_).end());
+            offset = (*asset_file_offsets_)[wrapper->handle_id];
+        }
+
+        // Filter duplicate calls to vkAllocateDescriptorSets for descriptor sets that were allocated by the same
+        // API call and reference the same parameter buffer.
+        if (processed.find(wrapper->create_parameters.get()) == processed.end())
+        {
+            processed.insert(wrapper->create_parameters.get());
+            if (wrapper->dirty)
```

Comment:
The wrapper can be not-dirty the next time the assets are dumped (the next time `WriteDescriptorSetStateWithAssetFile` is called) in the same run.

However I think the way I handle things here is a bit problematic because the `WriteFunctionCall` in line 1024 can write a single `vkAllocateDescriptorSets` for multiple descriptor sets and the second time the assets are dumped some can be dirty and other not-dirty.
I need to handle these cases properly.

Reply:
Was the single `vkAllocateDescriptorSets` call for multiple descriptor sets issue fixed?

Reply:
Yes. After some more thought I realized that this is handled correctly after all.